### PR TITLE
added default discovery port.  

### PR DESCRIPTION
**Pull request by GregHanson**
pilot-proxy and discovery currently have inconsistent default ports.  Discovery server is defaulting to 8080, and proxy talks to discovery:15003 by default.

### CODE REVIEW

**Comment by GregHanson** [comment]
Resolves https://github.com/istio/istio/issues/2168

**Comment by rshriram** [comment]
You are going to hate me for this, but it seems strange to overload model with all the defaults. Can we create a single defaults package with all the constants? Like we could put this in pkg/istiodefaults/pilot|mixer|security.go ? If that sounds bad, we could have the defaults.go in pilot as a single separate file that has all these hard coded choices.

File: install/kubernetes/templates/istio-pilot.yaml.tmpl [hunk 1]
```diff
@@ -90,7 +90,7 @@ spec:
       - name: discovery
         image: {PILOT_HUB}/pilot:{PILOT_TAG}
         imagePullPolicy: IfNotPresent
-        args: ["discovery", "-v", "2", "--admission-service", "istio-pilot"]
+        args: ["discovery", "-v", "2", "--port", "8080", "--admission-service", "istio-pilot"]
```
Comment (costinm) [inline install/kubernetes/templates/istio-pilot.yaml.tmpl]: Can we leave it as default port ? This is the 'insecure' port, that envoy forwards to, I don't see a good reason to change the default and then set a port to put it back to the old value.

File: pilot/cmd/pilot-discovery/main.go [hunk 2]
```diff
@@ -95,7 +96,7 @@ func init() {
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.ControllerOptions.DomainSuffix, "domain", "cluster.local",
 		"DNS domain suffix")
 
-	discoveryCmd.PersistentFlags().IntVar(&serverArgs.DiscoveryOptions.Port, "port", 8080,
+	discoveryCmd.PersistentFlags().IntVar(&serverArgs.DiscoveryOptions.Port, "port", model.DefaultDiscoveryPort,
```
Comment (costinm) [inline pilot/cmd/pilot-discovery/main.go]: Why ? The local port doesn't need to be exposed - envoy is listening on 15003.

File: pilot/pkg/model/context.go [hunk 3]
```diff
@@ -137,6 +137,9 @@ const (
 
 	// IngressKeyFilename is the ingress private key file name
 	IngressKeyFilename = "tls.key"
+
+	// DefaultDiscoveryPort is the default discovery port used by pilot and discovery
+	DefaultDiscoveryPort = 15003
```
Comment (costinm) [inline pilot/pkg/model/context.go]: Can you add a small comment about https/mtls ?

**Comment by costinm** [comment]
/hold

Adding a hold - a bit concerned with envoy config not updated and not sure I understand fully...

**Comment by GregHanson** [comment]
intended design is to always have an instance of envoy proxy running with a hardcoded config in front of pilot-discovery listening on port 15003 and forwarding calls to the discovery server on port 8080 to handle the possibility of mTLS between the pilot-agent instances and pilot-discovery.  Related design doc: https://docs.google.com/document/d/1LE7dbCQrhuvbxJuyV5sKlYLCtv0g9jfhteFIL06XCbI/edit#heading=h.x9snb54sjlu9

**Comment by GregHanson** [comment]
Hardcoded `envoy_pilot.json` and `envoy_pilot_auth.json` is located in `istio/pilot/docker`

**Pull request closed without merging**